### ISSUE TITLE
attention mask bugs

### ISSUE THREAD

**Issue by baiyimeng** [ZHZisZZ/dllm-trainer#15]
I saw your note that you “fixed attention mask bugs in modeling_lladamoe.py and modeling_llada.py” and recommend loading models with dllm.utils.get_tokenizer to ensure correctness.

Could you please clarify:

What exactly were the attention mask bugs?

How were they fixed?

Are there any specific implications for model training or inference if the fix is not applied?

**Comment by ZHZisZZ** [comment]
> I saw your note that you “fixed attention mask bugs in modeling_lladamoe.py and modeling_llada.py” and recommend loading models with dllm.utils.get_tokenizer to ensure correctness.

This is a typo, it has nothing to do with tokenizer; just use `import dllm` before `AutoModel.from_pretrained` then you will be using the modeling files from the `dllm`.

> Are there any specific implications for model training or inference if the fix is not applied?

Run this testcase with the original modeling file and you will know what i mean
https://github.com/ZHZisZZ/dllm/blob/main/scripts/tests/test_attention_mask.py#L12-L67. 

> How were they fixed?

See https://github.com/ZHZisZZ/dllm/pull/3

**Comment by ZHZisZZ** [comment]
Because LLaDA is trained to predict these padding eos tokens; LLaDA starts from a fixed length masks, and it needs to generate padding eos tokens to be able to generate variable length responses.

So the LLaDA training (pretrain+sft) does not require attention mask at all; you only need this for LLaDA batch inference.
